app/models/practice.rb

```diff
@@ -17,6 +21,30 @@ class Practice < ApplicationRecord
   attr_accessor :crop_x, :crop_y, :crop_w, :crop_h
   attr_accessor :practice_partner, :department
 
+  def clear_cache
+    cache_key = "searchable_practices"
+    Rails.cache.delete(cache_key)
+    Practice.searchable_practices
+  end
+
+  def clear_cache_on_update
+    if self.name_changed? ||
+        self.tagline_changed? ||
+        self.description_changed? ||
+        self.summary_changed? ||
+        self.initiating_facility_changed? ||
+        self.main_display_image_updated_at_changed? ||
+        self.published_changed?
+      clear_cache
+    end
+  end
+
+  def self.searchable_practices
```

Comment:
this is the reusable method that either stores or fetches from the cache.
if the cache from redis is nil, it runs the query in the block and caches it.
neato, huh?